diff --git a/INSTALL/superlu_timer.c b/INSTALL/superlu_timer.c
index 6ba07bf..9f79547 100644
--- a/INSTALL/superlu_timer.c
+++ b/INSTALL/superlu_timer.c
@@ -11,6 +11,7 @@
  *       compilation to choose the appropriate function.
  * </pre>
  */
+#define NO_TIMER
 
 
 #ifdef SUN 
@@ -56,6 +57,7 @@ double SuperLU_timer_()
     /* no sys/times.h on WIN32 */
     double tmp;
     tmp = 0.0;
+    int clocks_per_sec = CLK_TCK;
 #else
     struct tms use;
     double tmp;
diff --git a/Makefile b/Makefile
index d8d822c..fe7bc5d 100644
--- a/Makefile
+++ b/Makefile
@@ -18,16 +18,16 @@
 
 include make.inc
 
-all: install lib testing
+all: install lib
 
 lib: superlulib tmglib
 
 clean: cleanlib cleantesting
 
 install:
-	( cd INSTALL; $(MAKE) )
-#	( cd INSTALL; cp lsame.c ../SRC/; \
-#	  cp dlamch.c ../SRC/; cp slamch.c ../SRC/ )
+#	( cd INSTALL; $(MAKE) )
+	( cd INSTALL; cp lsame.c ../SRC/; \
+	  cp dlamch.c ../SRC/; cp slamch.c ../SRC/ )
 
 blaslib:
 	( cd CBLAS; $(MAKE) )
diff --git a/SRC/superlu_timer.c b/SRC/superlu_timer.c
index e888f08..783212d 100644
--- a/SRC/superlu_timer.c
+++ b/SRC/superlu_timer.c
@@ -11,7 +11,7 @@
  *       compilation to choose the appropriate function.
  * </pre>
  */
-
+#define NO_TIMER
 
 #ifdef SUN 
 /*
diff --git a/make.inc b/make.inc
index 2ffd9f4..f78e1b8 100644
--- a/make.inc
+++ b/make.inc
@@ -21,14 +21,14 @@ PLAT = _linux
 #
 #  The name of the libraries to be created/linked to
 #
-SuperLUroot	= $(HOME)/Codes/SuperLU/SuperLU_4.3
-SUPERLULIB   	= $(SuperLUroot)/lib/libsuperlu_4.3.a
+SuperLUroot	= $(SUPERLU_ROOT)
+SUPERLULIB   	= $(SuperLUroot)/SRC/libsuperlu_4.3.a
 TMGLIB       	= libtmglib.a
 
 ## BLASLIB   	= $(SuperLUroot)/lib/libblas.a
 
 BLASDEF 	= -DUSE_VENDOR_BLAS
-BLASLIB 	= -L$(HOME)/lib/GotoBLAS -lgoto
+BLASLIB 	= $(BLAS_LIB)
 
 ## ATLAS BLAS causes single-precision to fail
 #BLASLIB   	= -L/usr/lib/atlas -lblas
@@ -41,12 +41,10 @@ LIBS		= $(SUPERLULIB) $(BLASLIB)
 #  The archiver and the flag(s) to use when building archive (library)
 #  If your system has no ranlib, set RANLIB = echo.
 #
-ARCH         = ar
+ARCH         = $(AR)
 ARCHFLAGS    = cr
-RANLIB       = ranlib
 
-CC           = gcc
-CFLAGS       = -DPRNTlevel=0 -O3
+CFLAGS       = -DPRNTlevel=0 -O3 $(C_FLAGS)
 NOOPTS       = 
 FORTRAN	     = g77
 FFLAGS       = -O2
